File: Src/Engine/Editor/GUI/StatusBar.cpp

#include "StatusBar.h"

#include "Runtime/Render/2D/Render2D.h"
#include "Runtime/UI/GUI/WindowRootControl.h"

namespace SE::Editor
{
	StatusBar::StatusBar()
	{
		AutoFocus = false;
		AnchorPreset = AnchorPresets::HorizontalStretchBottom;
	}

	void StatusBar::Draw()
	{
		ContainerControl::Draw();

		Style* style = Style::Current;

		// Draw size grip
		WindowRootControl* window = TypeTryCast<WindowRootControl>(Root);
		if (window != nullptr && !window->IsMaximized())
			Render2D::DrawSprite(style->StatusBarSizeGrip, Rectangle(Width - 12, 10, 12, 12), style->Foreground);

		// Draw status text
		Render2D::RenderText(style->FontSmall, Text, Rectangle(4, 0, Width - 20, Height), TextColor, TextAlignment::Near, TextAlignment::Center);
	}
} // SE

File: Src/Engine/Editor/GUI/StatusBar.cs
// Managed editor GUI feature implementation.
using SE.GUI;
namespace SE.Editor.GUI
{
    public sealed class StatusBar : ContainerControl
    {
        public const float DefaultHeight = 22.0f;

        private readonly Label m_Label;
        private string m_Text = string.Empty;

        public StatusBar(float y, float width)
            : base(new Rectangle(0, y, width, DefaultHeight))
        {
            AutoFocus = false;
            m_Label = new Label(new Rectangle(4, 0, width - 20, DefaultHeight), string.Empty)
            {
                AutoFocus = false,
                Enabled = false,
            };
            SetBounds(0, y, width, DefaultHeight);
            AddChild(m_Label);
        }

        public string Text
        {
            get => m_Text;
            set
            {
                m_Text = value;
                m_Label.Text = value;
            }
        }

        public Color StatusColor
        {
            get => BackgroundColor;
            set => BackgroundColor = value;
        }

        public Color TextColor
        {
            get => m_Label.TextColor;
            set => m_Label.TextColor = value;
        }

        protected override void OnLayoutChildren()
        {
            m_Label.SetBounds(4, 0, Width - 20, Height);
        }
    }
}
